presentation/common/src/presentation-common/content/PropertyValueFormatter.ts

```diff
@@ -120,76 +119,92 @@ export class ContentPropertyValueFormatter {
         }
       : async (rawValue: number) => formatDouble(rawValue);
 
-    return this.formatValue(field.type, value, { doubleFormatter });
+    return this.formatValue(field, value, { doubleFormatter });
   }
 
-  private async formatValue(type: TypeDescription, value: Value, ctx?: { doubleFormatter: (raw: number) => Promise<string> }): Promise<DisplayValue> {
-    switch (type.valueFormat) {
-      case PropertyValueFormat.Primitive:
-        return this.formatPrimitiveValue(type, value, ctx);
-      case PropertyValueFormat.Array:
-        return this.formatArrayValue(type, value);
-      case PropertyValueFormat.Struct:
-        return this.formatStructValue(type, value);
+  private async formatValue(field: Field, value: Value, ctx?: { doubleFormatter: (raw: number) => Promise<string> }): Promise<DisplayValue> {
+    if (!field.isPropertiesField()) {
```

Comment:
This could break calculated properties - they're not based on specific property, but they are primitive and have type information. I think `formatPrimitiveValue` should take `Field` rather than `PropertiesField`.

Reply:
In this case `formatValue` should call `formatPrimitiveValue` when field is not propertyField instead of returning undefined right?

Reply:
yes, looks good now